### PR TITLE
Gateway 3381 - Audit Repository Logging for Doc Query

### PR DESCRIPTION
**Pull request by mlandis5am**
Modified logic to determine which hcid value should be used in the audit repo log depending on use case.

### CODE REVIEW

**Comment by bhumphrey** [comment]
A majority of the changes in this pull request relate to the gateway.properties and how to access it from test. Not 3381.

File: Product/Production/Services/DocumentQueryCore/src/test/java/gov/hhs/fha/nhinc/docquery/inbound/PassthroughInboundDocQueryTest.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
```
Comment (msweaver) [inline Product/Production/Services/DocumentQueryCore/src/test/java/gov/hhs/fha/nhinc/docquery/inbound/PassthroughInboundDocQueryTest.java]: This file needs to come out of this commit. This is organized in JIRA under GATEWAY-3371.

File: Product/Production/Services/DocumentQueryCore/src/test/java/gov/hhs/fha/nhinc/docquery/outbound/StandardOutboundDocQueryTest.java [hunk 2]
```diff
@@ -66,11 +94,11 @@ protected DocQueryAuditLog getAuditLogger() {
 
         verify(mockAuditLogger).auditDQRequest(eq(adhocQueryRequest), eq(assertion),
```
Comment (msweaver) [inline Product/Production/Services/DocumentQueryCore/src/test/java/gov/hhs/fha/nhinc/docquery/outbound/StandardOutboundDocQueryTest.java:95]: you are verifying the Entity (adapter) audits, but what about the Nhin?

File: Product/Production/Services/DocumentQueryCore/src/test/java/gov/hhs/fha/nhinc/docquery/inbound/PassthroughInboundDocQueryTest.java [hunk 3]
```diff
@@ -93,18 +127,31 @@ public void passthroughInboundDocQuery() {
 
         verify(mockAuditLogger).auditDQRequest(eq(request), eq(assertion),
                 eq(NhincConstants.AUDIT_LOG_INBOUND_DIRECTION), eq(NhincConstants.AUDIT_LOG_NHIN_INTERFACE),
-                any(String.class));
+                eq(sendingHcid));
 
         verify(mockAuditLogger).auditDQRequest(eq(request), eq(assertion),
```
Comment (msweaver) [inline Product/Production/Services/DocumentQueryCore/src/test/java/gov/hhs/fha/nhinc/docquery/inbound/PassthroughInboundDocQueryTest.java:98]: For GATEWAY-3371, the two Adapter interface audits should not be present. Try verify(mockAuditLogger, never())...etc

**Comment by msweaver** [comment]
Need unit tests for inbound.

File: Product/Production/Services/DocumentQueryCore/src/test/java/gov/hhs/fha/nhinc/docquery/outbound/StandardOutboundDocQueryTest.java (hunk 2 above)
Comment (mlandis5am) [inline Product/Production/Services/DocumentQueryCore/src/test/java/gov/hhs/fha/nhinc/docquery/outbound/StandardOutboundDocQueryTest.java:95]: These are here in the full file.

File: Product/Production/Services/DocumentQueryCore/src/test/java/gov/hhs/fha/nhinc/docquery/inbound/PassthroughInboundDocQueryTest.java (hunk 1 above)
Comment (mlandis5am) [inline Product/Production/Services/DocumentQueryCore/src/test/java/gov/hhs/fha/nhinc/docquery/inbound/PassthroughInboundDocQueryTest.java]: Pulling out - the changes here need to be pushed up to StandardInbound.

File: Product/Production/Services/DocumentQueryCore/src/main/java/gov/hhs/fha/nhinc/docquery/outbound/StandardOutboundDocQuery.java [hunk 4]
```diff
@@ -88,10 +88,8 @@ public AdhocQueryResponse respondingGatewayCrossGatewayQuery(AdhocQueryRequest a
         LOG.trace("EntityDocQueryOrchImpl.respondingGatewayCrossGatewayQuery...");
         AdhocQueryResponse response = null;
 
-        NhinTargetSystemType target = MessageGeneratorUtils.getInstance().convertFirstToNhinTargetSystemType(targets);
-        String targetHCID = getTargetHCID(target);
-
-        auditRequestFromAdapter(adhocQueryRequest, assertion, targetHCID);
+        String sendingHCID = HomeCommunityMap.getLocalHomeCommunityId();
```
Comment (jsmith2) [inline Product/Production/Services/DocumentQueryCore/src/main/java/gov/hhs/fha/nhinc/docquery/outbound/StandardOutboundDocQuery.java:91]: I don't think this is correct.  You're auditing the standard nhin outbound.  The hcid should be the value of the target, not the local hcid.

Comment (mlandis5am) [inline Product/Production/Services/DocumentQueryCore/src/main/java/gov/hhs/fha/nhinc/docquery/outbound/StandardOutboundDocQuery.java:91]: My understanding of the requirement was that the local hcid from gateway.properties is to be logged here.  the target hcid is logged in OutboundDocQueryStrategy for the NHIN entries.

Comment (msweaver) [inline Product/Production/Services/DocumentQueryCore/src/main/java/gov/hhs/fha/nhinc/docquery/outbound/StandardOutboundDocQuery.java:91]: If it's the adpater interfaces it should be the local hcid right?

**Pull request merged**